src/RoboSaverVirtualModule.sol

```diff
@@ -213,26 +266,26 @@ contract RoboSaverVirtualModule {
         uint256[] memory amountsOut = new uint256[](2);
         amountsOut[1] = _deficit;
 
-        // @todo do we need more math to calculate the exact amount of bpt to withdraw?
-        // @todo if not, explain why not in a @dev comment here
+        /// @dev Naive calculation of the `maxBPTAmountIn` based on the bpt rate and slippage %
+        uint256 maxBPTAmountIn = minAmountsOut[2] * MAX_BPS * 1e18 / (MAX_BPS - slippage) / BPT_STEUR_EURE.getRate();
```

Comment:
here im not sure the order of the multiplication of `MAX_BPS` divided later by `MAX_BPS - slippage` is correct

would not that be the equivalent of dividing 10k / 9.8k increasing the `deficit` amount by ~`1.02`?

i.e: `deficit * 1.02 / rate`

Reply:
or is it like to give a bit of a buffer on how much bpt can be wd w/ that 1.02 multiplier to ensure that the deficit is covered?

Reply:
yes; we allow 2% more bpt to be burned in case that that is needed to withdraw at least `_deficit - 1`

Reply:
got it. to ensure that the min is always meet w/ the surplus of bpt being burned ✅ 

only L im doubting now it is if really we want to have as `minAmountsOut[EURE_TOKEN_BPT_INDEX] = _deficit - 1;` specially w/ the surplus in bpt being burned, maybe better to keep it strict at:

```
minAmountsOut[EURE_TOKEN_BPT_INDEX] = _deficit;
```

in this way we ensure that precise deficit is always met imo, instead of allowing loosing that 1 wei somehow

Reply:
yea this makes sense imo. i think you added the `- 1` due to an edge case at some point, but i guess that is not applicable anymore now and replaced by the slippage calc. 81620f67c620cf9502cab5fb207b09e7561d2879